src/main/java/command/DeleteCommand.java

```diff
@@ -0,0 +1,43 @@
+package command;
+
+import duke.DukeException;
+import duke.Storage;
+import duke.TaskList;
+import duke.Ui;
+import task.Task;
+
+import static java.lang.String.format;
+
+/**
+ * Command to delete a task.
+ */
+public class DeleteCommand extends Command {
+    private int index;
+
+    /**
+     * Command to delete a task.
+     *
+     * @param input the task number in string.
+     * @throws DukeException If the task number is of invalid format.
+     */
+    public DeleteCommand(String input) throws DukeException {
+        try {
+            this.index = Integer.parseInt(input);
+            this.setExit(false);
```

Comment:
Calling `setExit(false)` in all command constructors is a lot of work - consider having the field set in the constructor of the abstract class (which will be invoked by the child classes) instead.